lib/main.js

```diff
@@ -189,8 +189,16 @@ function main() {
             if (enableOIDC) {
                 commonArgs = commonArgs.concat("--federated-token", federatedToken);
             }
-            else {
-                commonArgs = commonArgs.concat("-p", servicePrincipalKey);
+            else {               
+                    if(servicePrincipalKey.startsWith("-"))
+                    {
+                        commonArgs = commonArgs.concat("-p", "\"" + servicePrincipalKey + "\"");
```

Comment:
Why are we having the separate condition for with and without hyphen(-). Can't we make it generic? @AjayKaleMS